### DISCUSSION TITLE
Option to disable AutoExtendClaimTask, in some worlds.

### DISCUSSION THREAD

**Discussion by MrZeLee** [GriefPrevention/GriefPrevention#673] [Ideas]
As an admin, I would like the option to disable AutoExtendClaimTask, in some worlds.
I think is a pretty easy feature to implement, adding to the config file;
AutoExtendClaim:
world: true
world_nether: false
world_the-end:false

The main goal is to when players are creating paths in nether and end, they don't block all the area for the other players to build.

PS:. In the resizing part, instead of checking if you are in the exact y block, just use this:

-> If shovel is in main hand &&  ClickedY > claim.greatterY ->  claim.greatterY = ClickedY
-> If shovel is in main hand &&  ClickedY < claim.lowerY ->  claim.lowerY = ClickedY

-> If shovel is in off hand &&  ClickedY < claim.greatterY ->  claim.greatterY = ClickedY
-> If shovel is in off hand && SHIFT &&  ClickedY > claim.lowerY ->  claim.lowerY = ClickedY

(Maybe better use a command to toogle between resizing Y to increase, or Y to decrease Area)

**Comment by MrZeLee** [comment]
I tried to implement some features that really take y into consideration, on the overlaps and protected blocks.
Ex.: If block inside xz claimed area, but y > greattercornerY the block can be destroyed.

This allows to create claims in height (y axis)

But destroys the grief of water flowing, and pistons I think

**Comment by RoboMWM** [comment]
You can set the minimum y depth claims can go to in the config.

But from your second comment it now sounds like you're trying to implement 3D claims e.g. #269

**Comment by MrZeLee** [comment]
I tryed to implement 3D claims and it works, if you want I will send you the code.
It doesn't  have the prevention of pistons vertically and watter falling from an upper claim, but at least the other players don't have the ability to break blocks in the claim.

**Comment by RoboMWM** [comment]
You should release your code as an addon then, that way if server admins don't care for those protections yet want 3D claims, then they can use your addon.

**Comment by MrZeLee** [comment]
can you send me some information, on how could I start to create an addon to GriefPreventions

**Comment by RoboMWM** [comment]
Easiest is probably to look at other GP addons. Basically you get the plugin's instance from the Bukkit plugin manager whenever you need to get any information from GP.

Whenever the v20 update is finished this process will be easier.
